src/ui/NoNameFacet/NoNameFacetHeader/NoNameFacetHeader.ts

```diff
@@ -0,0 +1,76 @@
+import { $$ } from '../../../utils/Dom';
+import { l } from '../../../strings/Strings';
+import { SVGIcons } from '../../../utils/SVGIcons';
+import { SVGDom } from '../../../utils/SVGDom';
+import { INoNameFacetOptions } from '../NoNameFacetOptions';
+import { NoNameFacetHeaderClear } from './NoNameFacetHeaderClear';
+import { NoNameFacetHeaderCollapseToggle } from './NoNameFacetHeaderCollapseToggle';
+import { NoNameFacetHeaderOperatorToggle } from './NoNameFacetHeaderOperatorToggle';
+
+export interface INoNameFacetHeaderOptions {
+  rootFacetOptions: INoNameFacetOptions;
+}
+
+export class NoNameFacetHeader {
+  public element: HTMLElement;
+  private waitAnimationElement: HTMLElement;
+  private clearButton: NoNameFacetHeaderClear;
+  private operatorToggle: NoNameFacetHeaderOperatorToggle;
+  private collapseToggle: NoNameFacetHeaderCollapseToggle;
+
+  constructor(private options: INoNameFacetHeaderOptions) {
+    this.element = $$('div', { className: 'coveo-facet-header' }).el;
+    $$(this.element).append(this.createTitleSection());
+    $$(this.element).append(this.createSettingsSection());
+  }
+
+  private createTitleSection() {
+    const section = $$('div', { className: 'coveo-facet-header-title-section' });
+
+    section.append(this.createTitle());
+
+    this.waitAnimationElement = this.createWaitAnimation();
+    section.append(this.waitAnimationElement);
+    this.toggleLoading(false);
+
+    return section.el;
+  }
+
+  private createSettingsSection() {
+    const { useAnd, isCollapsed, enableCollapse, enableOperator } = this.options.rootFacetOptions;
+    const section = $$('div', { className: 'coveo-facet-header-settings-section' });
+
+    this.clearButton = new NoNameFacetHeaderClear();
+    section.append(this.clearButton.create());
+
+    this.operatorToggle = new NoNameFacetHeaderOperatorToggle({ useAnd });
+    enableOperator && section.append(this.operatorToggle.create());
+
+    this.collapseToggle = new NoNameFacetHeaderCollapseToggle({ isCollapsed });
+    enableCollapse && section.append(this.collapseToggle.create());
+
+    return section.el;
+  }
+
+  private createTitle() {
+    const title = $$('div', { className: 'coveo-facet-header-title' }, this.options.rootFacetOptions.title);
+    title.setAttribute('role', 'heading');
+    title.setAttribute('aria-level', '2');
```

Comment:
I would start thinking about adding tests in `./accessibilityTest/`

Reply:
sure, but should we still test it here?

Reply:
Not sure what you mean ? If you mean tests can be added at a later time/another PR, sure thing.

Just a "heads up" to not forget about it :)

Reply:
alright!